08-walkthrough-3.Rmd

```diff
@@ -121,67 +113,115 @@ d <- d %>%
 
 ```{r}
+#First, we set a seed to ensure the reproducibility of our data partition.
+set.seed(62020)
+
+#we create a new object called trainIndex that will take 80 percent of the data
 trainIndex <- createDataPartition(d$final_grade,
                                   p = .8, 
                                   list = FALSE,
                                   times = 1)
 
-d_train <- d[ trainIndex,] # can we make this tidy w/ filter
-d_test <- d[-trainIndex,]
+
+#We add a new variable to our dataset, temporarily:
+    #this will let us select our rows according to their row number
+    #we populate the rows with the numbers 1:499, in order
+
+d <- d %>% 
+    mutate(temp_id = 1:499)
+
+#we filter our dataset so that we get only the 
+    #rows indicated by our "trainIndex" vector
+d_train <- filter(d, temp_id %in% trainIndex)
+
+#we filter our dataset in a different way so that we get only  the rows NOT in our "trainIndex" vector
+    #adding the ! before the temp_id variable achieves the opposite of what we did in the line of code above
+
+d_test <- filter(d, !temp_id %in% trainIndex)
+
+#We delete the temp_id variable from (1) the original data, (2) the portion of the original data we marked as training, and (3) the portion of the original data we marked as testing, as we no longer need that variable
+
+d <- select(d, -temp_id)
+d_train <- select(d_train, - temp_id)
+d_test <- select(d_test, -temp_id)
 ```
 
 Finally, we will estimate the models.
 
-Here, we will use the train function, passing *all* of the variables in the data frame (except for the outcome, or dependent variable, `final_grade`) as predictors. NOte that you can read more about the specific random forest implementation chosen [here](http://topepo.github.io/caret/train-models-by-tag.html#random-forest).
+Here, we will use the train function, passing *all* of the variables in the data frame (except for the outcome, or dependent variable, `final_grade`) as predictors. Note that you can read more about the specific random forest implementation chosen [here](http://topepo.github.io/caret/train-models-by-tag.html#random-forest). To specify that we want to predict the outcome using every variable except the outcome itself, we use the formulation (outcome ~ .,). R interprets this code as: predict the outcome using all the variables except outcome itself. The outcome always comes before the `~`, and the `.` that we see after the `~` means that we want to use all the rest of the variables. An alternative specification of this model would be to write (outcome ~ predictor1, predictor2). Anything that follows the `~` and precedes the comma is treated as predictors of the outcome.
+
+
+Here, we set the seed again, to ensure that our analysis is reproducible. This step of setting the seed is especially important due to the "random" elements of random forest, because it's likely that the findings would change (just slightly) if the seed were not set.
 
```

Comment:
it might be helpful to reiterate what our outcomes and predictors are

Reply:
Great point. Will do